activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -323,7 +323,9 @@ def supports_views?
       end
 
       def supports_datetime_with_precision?
-        (version[0] == 5 && version[1] >= 6) || version[0] >= 6
+        version[0] == 5 && version[1] == 6 && version[2] >= 4 ||
+          version[0] == 5 && version[1] >= 7 ||
+          version[0] >= 6
```

Comment:
Can we wrap this in a `Version` object which knows how to do comparisons so that this can be written as follows? The current code makes my head hurt.

```
version >= Version.new(5, 6, 4)
```